splice_spark/src/main/spark2.3/com/splicemachine/spark/splicemachine/SplicemachineContext.scala

```diff
@@ -356,28 +365,28 @@ class SplicemachineContext(options: Map[String, String]) extends Serializable {
     path.toString
   }
 
-  def getRandomName(): String = {
+  private[this] def getRandomName(): String = {
     val name = new Array[Byte](32)
     new SecureRandom().nextBytes(name)
     Bytes.toHex(name)+"-"+System.nanoTime()
   }
 
   /**
-    * SQL to Dataframe translation.  (Lazy)
+    * SQL to Dataset translation.  (Lazy)
     *
-    * @param sql
-    * @return
+    * @param sql SQL query
+    * @return Dataset[Row] with the result of the query
     */
   def df(sql: String): Dataset[Row] = {
     SparkUtils.resultSetToDF(internalConnection.createStatement().executeQuery(sql));
   }
 
   /**
-    * SQL to Dataframe translation.  (Lazy)
+    * SQL to Dataset translation.  (Lazy)
```

Comment:
The comments get worse every change :( `Dataset` is lazy by design so there's no point mentioning it (it's implied). Rather than describing what's included in the signature, I'd rather focus on why a developer would want to use it.